addons/account_tax_cash_basis/tests/test_account_tax_cash_basis.py

```diff
@@ -0,0 +1,160 @@
+            'payment_method_id': self.pay_method.id,
+            'partner_id': invoice.partner_id.id,
+            'currency_id': self.currency_usd_id,
+            'partner_type': 'customer'})
+        payment.post()
+
+    def create_account_tax_er(self):
+        """This account is created to use like cash basis account and only
+        it will be filled when there is payment
+        """
+        account_ter = self.account_model.create({
+            'name': 'Tax effectively receivable',
+            'code': '11111101',
+            'user_type_id': self.user_type_id.id,
+        })
+        return account_ter
+
+    def create_invoice_usd(self, date_invoice):
+        """Create to invoice in given date
+        :param date_invoice: the date invoice and this date will be used to get
+        the currency rate"""
+
+        tax = self.env['account.tax'].create({
+            'name': 'Tax 16.0',
+            'amount': 16.0,
+            'amount_type': 'percent',
+            'account_id': self.account_tax_receivable.id,
+            'use_cash_basis': True,
+            'cash_basis_account': self.account_tax_cash_basis.id
+        })
+
+        invoice = self.account_invoice_model.create({
+            'partner_id': self.partner_agrolait_id,
+            'reference_type': 'none',
+            'currency_id': self.currency_usd_id,
+            'name': 'invoice to client',
+            'account_id': self.account_rcv.id,
+            'type': 'out_invoice',
+            'date_invoice': date_invoice,
+            })
+        self.account_invoice_line_model.create({
+            'product_id': self.product.id,
+            'quantity': 1,
+            'price_unit': 100,
+            'invoice_id': invoice.id,
+            'name': 'product that cost 100',
+            'account_id': self.account_model.search(
+                [('user_type_id', '=', self.env.ref(
+                    'account.data_account_type_revenue').id)], limit=1).id,
+            'invoice_line_tax_ids': [(4, tax.ids)]
+        })
+
+        # validate invoice
+        invoice.compute_taxes()
+        invoice.action_invoice_open()
+        return invoice
+
+    def test_cash_basis_tax_rate_invoice_greater_than_payment(self):
+        """Test to validate tax effectively receivable
+
+        Case   |   rate | expected tax effectively receivable
+       --------|--------|-----------------------------------
+       Invoice | 1.2834 | Not yet created
+       Payment | 1.5289 | 10.47
+        """
+        invoice_id = self.create_invoice_usd(time.strftime('%Y') + '-01-01')
+        self.create_payment(invoice_id, time.strftime('%Y') + '-07-15', 116)
+        move_ter = self.account_move_line_model.search(
+            [('account_id', '=', self.account_tax_cash_basis.id)])
+        ter_balance = (
+            sum(move_ter.mapped('credit')) - sum(move_ter.mapped('debit')))
+        self.assertEquals(
+            float_round(ter_balance, precision_digits=self.precision), 10.47)
```

Comment:
In this case, the following entries are created:

INVOICE:

account|debit|credit
-----|-----|-----
Receivable|90.39|
Fixed||12.47
Product||77.92
Tot.|90.39|90.39

BANK:

account|debit|credit
-----|-----|-----
Bank|75.87|
Receivable||77.87
Tot.|77.87|77.87

BANK EXCHANGE:

account|debit|credit
-----|-----|-----
Exchange|14.52|
Receivable||14.52
Tot.|14.52|14.52

CASH:

account|debit|credit
-----|-----|-----
Fixed|12.47|
Inter.||12.47
Product|77.92|
Product||77.92
Tot.|90.39|90.39

CASH EXCHANGE:

account|debit|credit
-----|-----|-----
Fixed|2.0|
Inter.||2.0
Product|12.52|
Product||12.52
Tot.|14.52|14.52

That leads to following balances:

FIXED:

debit|credit
-----|-----
2.0|0.0

PRODUCT:

debit|credit
-----|-----
0.0|77.92

EXCHANGE:

debit|credit
-----|-----
14.52|0.0

BANK:

debit|credit
-----|-----
75.87|0.0

INTER.:

debit|credit
-----|-----
0.0|14.47

I agree there is something wrong here (in INTER./FIXED). In my opinion, the error comes from the CASH EXCHANGE that should be:

 account|debit|credit
-----|-----|-----
Inter.|2.0|
Fixed||2.0
Product|12.52|
Product||12.52
Tot.|14.52|14.52

to change FIXED:

debit|credit
-----|-----
0.0|2.0

and having INTER. like:

debit|credit
-----|-----
2.0|12.47

... that seems correct according to this test.

My question is about FIXED, it is correct to have 2.0 remaining credit in this case?